presto-main/src/main/java/com/facebook/presto/util/GraphvizPrinter.java

```diff
@@ -185,6 +185,22 @@ public static String printDistributed(SubPlan plan, Session session, FunctionAnd
 
         return output.toString();
     }
+    public static String printDistributedFromFragments(List<PlanFragment> allFragments, Session session, FunctionAndTypeManager functionAndTypeManager)
+    {
+        PlanNodeIdGenerator idGenerator = new PlanNodeIdGenerator();
+        Map<PlanFragmentId, PlanFragment> fragmentsById = Maps.uniqueIndex(allFragments, PlanFragment::getId);
+
+        StringBuilder output = new StringBuilder();
+        output.append("digraph distributed_plan {\n");
+
+        for (PlanFragment planFragment : allFragments) {
```

Comment:
Should we sort `allFragments` by id so that the fragments are printed in order ? This won't impact rendered svg's, but will keep the text easier to read.

Reply:
good point, in fact the sort should be done before it's passed in